rust/src/lib/ifaces/vxlan.rs

```diff
@@ -48,18 +48,26 @@ impl VxlanInterface {
     }
 
     pub(crate) fn parent(&self) -> Option<&str> {
-        self.vxlan.as_ref().map(|cfg| cfg.base_iface.as_str())
+        self.vxlan
+            .as_ref()
+            .and_then(|cfg| cfg.base_iface.as_ref())
+            .map(|base_iface| base_iface.as_str())
     }
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
 #[non_exhaustive]
 pub struct VxlanConfig {
-    pub base_iface: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub base_iface: Option<String>,
```

Comment:
This is breaking rust API which we have no plan on releasing nmstate 3.x.

Please use `serde(default)` to store the undefined `base_iface` as emptry.